<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CornDetectionMainWindow</class>
 <widget class="QMainWindow" name="CornDetectionMainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1200</width>
    <height>819</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Aplikasi Deteksi Jagung - Machine Learning</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="mainLayout">
    <item>
     <widget class="QWidget" name="leftPanel" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>1</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <layout class="QVBoxLayout" name="leftLayout">
       <item>
        <widget class="QLabel" name="titleLabel">
         <property name="styleSheet">
          <string notr="true">font-size: 24px;
font-weight: bold;
color: #2c3e50;
padding: 20px;
background-color: #ecf0f1;
border-radius: 10px;
margin-bottom: 10px;</string>
         </property>
         <property name="text">
          <string>DETEKSI JAGUNG</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Muhammad Rifkyandryan Rustanto - F1D022077</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="imageLabel">
         <property name="minimumSize">
          <size>
           <width>300</width>
           <height>300</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true">border: 2px dashed #cccccc;
border-radius: 10px;
background-color: #f9f9f9;</string>
         </property>
         <property name="text">
          <string>Pilih gambar jagung untuk dianalisis</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QVBoxLayout" name="buttonLayout">
         <item>
          <widget class="QPushButton" name="btnSelectImage">
           <property name="styleSheet">
            <string notr="true">background-color: #3498db;
color: white;
border: none;
padding: 10px;
font-size: 14px;
border-radius: 5px;
font-weight: bold;</string>
           </property>
           <property name="text">
            <string>Pilih Gambar</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="modelGroup">
           <property name="styleSheet">
            <string notr="true">font-weight: bold;
border: 2px solid #bdc3c7;
border-radius: 8px;
margin-top: 10px;
padding-top: 10px;
background-color: white;</string>
           </property>
           <property name="title">
            <string>Pilih Model</string>
           </property>
           <layout class="QGridLayout" name="modelLayout">
            <item row="0" column="1">
             <widget class="QComboBox" name="modelCombo">
              <property name="styleSheet">
               <string notr="true">padding: 8px;
border: 2px solid #ddd;
border-radius: 4px;
background-color: white;</string>
              </property>
              <item>
               <property name="text">
                <string>COLOR_RF (Akurasi: 0.8247)</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>LBP_RF (Akurasi: 0.7731)</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>GLCM_RF (Akurasi: 0.7325)</string>
               </property>
              </item>
             </widget>
            </item>
            <item row="0" column="0">
             <widget class="QLabel" name="modelLabel">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Model:</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="btnPredict">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="styleSheet">
            <string notr="true">background-color: #3498db;
color: white;
border: none;
padding: 10px;
font-size: 14px;
border-radius: 5px;
font-weight: bold;</string>
           </property>
           <property name="text">
            <string>Prediksi</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QGroupBox" name="resultsGroup">
         <property name="styleSheet">
          <string notr="true">font-weight: bold;
border: 2px solid #bdc3c7;
border-radius: 8px;
margin-top: 10px;
padding-top: 10px;
background-color: white;</string>
         </property>
         <property name="title">
          <string>Hasil Prediksi</string>
         </property>
         <layout class="QFormLayout" name="resultsLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="resultTextLabel">
            <property name="text">
             <string>Hasil:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLabel" name="resultLabel">
            <property name="text">
             <string>-</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="confidenceTextLabel">
            <property name="text">
             <string>Confidence:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLabel" name="confidenceLabel">
            <property name="text">
             <string>-</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="processingTimeTextLabel">
            <property name="text">
             <string>Waktu Proses:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QLabel" name="processingTimeLabel">
            <property name="text">
             <string>-</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="leftSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QWidget" name="rightPanel" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>2</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <layout class="QVBoxLayout" name="rightLayout">
       <item>
        <widget class="QGroupBox" name="dbGroup">
         <property name="styleSheet">
          <string notr="true">font-weight: bold;
border: 2px solid #bdc3c7;
border-radius: 8px;
margin-top: 10px;
padding-top: 10px;
background-color: white;</string>
         </property>
         <property name="title">
          <string>Riwayat Prediksi</string>
         </property>
         <layout class="QVBoxLayout" name="dbLayout">
          <item>
           <layout class="QHBoxLayout" name="searchLayout">
            <item>
             <widget class="QLabel" name="searchLabel">
              <property name="text">
               <string>Cari:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="searchInput">
              <property name="styleSheet">
               <string notr="true">padding: 8px;
border: 2px solid #ddd;
border-radius: 4px;
background-color: white;</string>
              </property>
              <property name="placeholderText">
               <string>Cari berdasarkan nama file...</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="filterLabel">
              <property name="text">
               <string>Filter:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="filterCombo">
              <property name="styleSheet">
               <string notr="true">padding: 8px;
border: 2px solid #ddd;
border-radius: 4px;
background-color: white;</string>
              </property>
              <item>
               <property name="text">
                <string>Semua</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Daun Jagung Baik</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Daun Jagung Sakit</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Bukan Daun Jagung</string>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QTableWidget" name="table">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>1</verstretch>
          </sizepolicy>
         </property>
         <property name="styleSheet">
          <string notr="true">gridline-color: #ddd;
background-color: white;
alternate-background-color: #f8f9fa;
selection-background-color: #3498db;</string>
         </property>
         <property name="alternatingRowColors">
          <bool>true</bool>
         </property>
         <property name="selectionBehavior">
          <enum>QAbstractItemView::SelectRows</enum>
         </property>
         <column>
          <property name="text">
           <string>ID</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Nama File</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Hasil Prediksi</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Model</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Tanggal</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Confidence</string>
          </property>
         </column>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="dbButtonLayout">
         <item>
          <widget class="QPushButton" name="btnRefresh">
           <property name="styleSheet">
            <string notr="true">background-color: #3498db;
color: white;
border: none;
padding: 10px;
font-size: 14px;
border-radius: 5px;
font-weight: bold;</string>
           </property>
           <property name="text">
            <string>Refresh</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="btnDelete">
           <property name="styleSheet">
            <string notr="true">background-color: #3498db;
color: white;
border: none;
padding: 10px;
font-size: 14px;
border-radius: 5px;
font-weight: bold;</string>
           </property>
           <property name="text">
            <string>Hapus</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="btnExportCSV">
           <property name="styleSheet">
            <string notr="true">background-color: #3498db;
color: white;
border: none;
padding: 10px;
font-size: 14px;
border-radius: 5px;
font-weight: bold;</string>
           </property>
           <property name="text">
            <string>Export CSV</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="btnExportPDF">
           <property name="styleSheet">
            <string notr="true">background-color: #3498db;
color: white;
border: none;
padding: 10px;
font-size: 14px;
border-radius: 5px;
font-weight: bold;</string>
           </property>
           <property name="text">
            <string>Export PDF</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="dbButtonSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QGroupBox" name="statsGroup">
         <property name="styleSheet">
          <string notr="true">font-weight: bold;
border: 2px solid #bdc3c7;
border-radius: 8px;
margin-top: 10px;
padding-top: 10px;
background-color: white;</string>
         </property>
         <property name="title">
          <string>Statistik Model</string>
         </property>
         <layout class="QGridLayout" name="statsLayout">
          <item row="1" column="1">
           <widget class="QLabel" name="colorRFAkurasi">
            <property name="styleSheet">
             <string notr="true">padding: 3px; border: 1px solid #ddd;</string>
            </property>
            <property name="text">
             <string>82.47%</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="colorRF">
            <property name="styleSheet">
             <string notr="true">padding: 3px; border: 1px solid #ddd;</string>
            </property>
            <property name="text">
             <string>COLOR RF</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QLabel" name="statsPresisi">
            <property name="styleSheet">
             <string notr="true">font-weight: bold; background-color: #e0e0e0; padding: 5px;</string>
            </property>
            <property name="text">
             <string>Presisi</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item row="0" column="3">
           <widget class="QLabel" name="statsRecall">
            <property name="styleSheet">
             <string notr="true">font-weight: bold; background-color: #e0e0e0; padding: 5px;</string>
            </property>
            <property name="text">
             <string>Recall</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="statsModel">
            <property name="styleSheet">
             <string notr="true">font-weight: bold; background-color: #e0e0e0; padding: 5px;</string>
            </property>
            <property name="text">
             <string>Model</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLabel" name="statsAkurasi">
            <property name="styleSheet">
             <string notr="true">font-weight: bold; background-color: #e0e0e0; padding: 5px;</string>
            </property>
            <property name="text">
             <string>Akurasi</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="lbpRF">
            <property name="styleSheet">
             <string notr="true">padding: 3px; border: 1px solid #ddd;</string>
            </property>
            <property name="text">
             <string>LBP RF</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QLabel" name="colorRFPresisi">
            <property name="styleSheet">
             <string notr="true">padding: 3px; border: 1px solid #ddd;</string>
            </property>
            <property name="text">
             <string>80.66%</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item row="1" column="3">
           <widget class="QLabel" name="colorRFRecall">
            <property name="styleSheet">
             <string notr="true">padding: 3px; border: 1px solid #ddd;</string>
            </property>
            <property name="text">
             <string>77.89%</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item row="2" column="2">
           <widget class="QLabel" name="lbpRFPresisi">
            <property name="styleSheet">
             <string notr="true">padding: 3px; border: 1px solid #ddd;</string>
            </property>
            <property name="text">
             <string>73.80%</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item row="2" column="3">
           <widget class="QLabel" name="lbpRFRecall">
            <property name="styleSheet">
             <string notr="true">padding: 3px; border: 1px solid #ddd;</string>
            </property>
            <property name="text">
             <string>66.95%</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QLabel" name="lbpRFAkurasi">
            <property name="styleSheet">
             <string notr="true">padding: 3px; border: 1px solid #ddd;</string>
            </property>
            <property name="text">
             <string>77.31%</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="glcmRF">
            <property name="styleSheet">
             <string notr="true">padding: 3px; border: 1px solid #ddd;</string>
            </property>
            <property name="text">
             <string>GLCM RF</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item row="3" column="3">
           <widget class="QLabel" name="glcmRFRecall">
            <property name="styleSheet">
             <string notr="true">padding: 3px; border: 1px solid #ddd;</string>
            </property>
            <property name="text">
             <string>63.71%</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QLabel" name="glcmRFAkurasi">
            <property name="styleSheet">
             <string notr="true">padding: 3px; border: 1px solid #ddd;</string>
            </property>
            <property name="text">
             <string>73.25%</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item row="3" column="2">
           <widget class="QLabel" name="glcmRFPresisi">
            <property name="styleSheet">
             <string notr="true">padding: 3px; border: 1px solid #ddd;</string>
            </property>
            <property name="text">
             <string>67.46%</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1200</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <widget class="QMenu" name="menuExport">
     <property name="title">
      <string>Export</string>
     </property>
     <addaction name="actionExportCSV"/>
     <addaction name="actionExportPDF"/>
    </widget>
    <addaction name="actionOpenImage"/>
    <addaction name="menuExport"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
    <addaction name="actionRefresh"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuView"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionOpenImage">
   <property name="text">
    <string>Buka Gambar</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionExportCSV">
   <property name="text">
    <string>Export ke CSV</string>
   </property>
  </action>
  <action name="actionExportPDF">
   <property name="text">
    <string>Export ke PDF</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Keluar</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionRefresh">
   <property name="text">
    <string>Refresh Data</string>
   </property>
   <property name="shortcut">
    <string>F5</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>Tentang</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
